### PR TITLE
21 add endpoint to check sync

### LINKED ISSUE #21: Add endpoint to check syncronization
- [x] I have read the [Contributing](https://github.com/swapper-org/NodeChain/blob/master/CONTRIBUTING.md) file of the documentation.
- [x] I have searched the [issues](https://github.com/swapper-org/NodeChain/issues) (including closed ones) and believe that this is not a duplicate.
- [x] I am willing to lend a hand to help implement this feature. 

### Feature Description

It would be a good idea to implement an endpoint to check if any blockchain is syncing.

A good response template could be:
```
coin: ETH
syncing: true
<!-- If syncing -->
startingBlock: '0x384',
currentBlock: '0x386',
highestBlock: '0x454'

```

### PR DESCRIPTION
**Pull request by phoenix-web3**
# Description

Fixes #21  (issue)

## Dependencies (if any)

- Fix ethereumgo connection bug
- Add endpoint to check synchronization in ETH, BTC and BCH.
- Add BCH constants to make it similar to BTC

## Type of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [x] **This change requires a documentation update**

# How Has This Been Tested?

### ETH SYNC

```sh
curl -X POST -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0","id":1, "method":"syncing","params":{}}' http://localhost:60001/rpc
{"id": 1, "jsonrpc": "2.0", "result": {"syncing": true, "syncPercentage": "6.905658629231632%", "currentBlockIndex": 755778, "latestBlockIndex": 10944329}}
```

### BTC SYNC
```sh
~$ curl -X POST -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0","id":1, "method":"syncing","params":{}}' http://localhost:60001/rpc
{"id": 1, "jsonrpc": "2.0", "result": {"syncing": true, "syncPercentage": "5.732548291803841%", "currentBlockIndex": 352747, "latestBlockIndex": 2101201}}
```

**Test Configuration**:
* Operating system (output of `cat /etc/os-release`):
```sh
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
* Kernel version (output of `uname -sr`): Linux 4.15.0-143-generic
* Architecture (output of `uname -m`): x86_64

# Related PR or Docs PR

[Docs PR related](https://github.com/swapper-org/NodeChain-docs/pull/1) 
Other PR related # 

# Good practices to consider

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by 0xp3gasus** [comment]
Where is the RPC Endpoint to check sync in BTC?

**Comment by phoenix-web3** [comment]
> Where is the RPC Endpoint to check sync in BTC?

Last commit

File: docker-compose/mainnet/eth.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: "3"
 services:
   ethereumgo:
     image: ethereum/client-go:latest
-    command: --http --http.addr=ethereumgo --cache=1024 --http.port=8545 --syncmode=full --gcmode=archive --ws --ws.addr=ethereumgo --ws.port=8546 --ws.origins=connector
+    command: --http --http.addr=ethereumgo --cache=1024 --http.port=8545 --syncmode=full --gcmode=archive --ws --ws.addr=ethereumgo --ws.port=8546 --ws.origins=connector --http.vhosts=ethereumgo
```
Comment (0xp3gasus) [inline docker-compose/mainnet/eth.yml:5]: According to parameter definition: 
`--http.vhosts value                 Comma separated list of virtual hostnames from which to accept requests (server enforced). Accepts '*' wildcard. (default: "localhost")`

Why did you set ethereumgo as value? Why should not be Connector?

File: docker-compose/testnet/eth.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: "3"
 services:
   ethereumgo:
     image: ethereum/client-go:latest
-    command: --http --http.addr=ethereumgo --ropsten --cache=1024 --http.port=8545 --syncmode=full --gcmode=archive --ws --ws.addr=ethereumgo --ws.port=8546 --ws.origins=connector
+    command: --http --http.addr=ethereumgo --ropsten --cache=1024 --http.port=8545 --syncmode=full --gcmode=archive --ws --ws.addr=ethereumgo --ws.port=8546 --ws.origins=connector --http.vhosts=ethereumgo
```
Comment (0xp3gasus) [inline docker-compose/testnet/eth.yml:5]: According to parameter definition:
`--http.vhosts value Comma separated list of virtual hostnames from which to accept requests (server enforced). Accepts '*' wildcard. (default: "localhost")`

Why did you set ethereumgo as value? Why should not be Connector?

**Review by 0xp3gasus — CHANGES_REQUESTED** [review]
Please, review comments. Thank you

File: Connector/bch/rpcschemas/syncing_response.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        "type": "string"
+    },
+    "currentBlock":{
+        "type": "string"
+    },
+    "latestBlock":{
+        "type": "string"
+    }
+},
+"required": [
```
Comment (0xp3gasus) [inline Connector/bch/rpcschemas/syncing_response.json]: Use correct tabs

File: Connector/btc/rpcschemas/syncing_response.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        "type": "string"
+    },
+    "currentBlock":{
+        "type": "string"
+    },
+    "latestBlock":{
+        "type": "string"
+    }
+},
+"required": [
```
Comment (0xp3gasus) [inline Connector/btc/rpcschemas/syncing_response.json]: Use correct tabs

Comment (phoenix-web3) [reply Connector/btc/rpcschemas/syncing_response.json]: Resolved in last commit

File: Connector/bch/rpcschemas/syncing_response.json (hunk 3 above)
Comment (phoenix-web3) [reply Connector/bch/rpcschemas/syncing_response.json]: Resolved in last commit

File: docker-compose/testnet/eth.yml (hunk 2 above)
Comment (phoenix-web3) [reply docker-compose/testnet/eth.yml:5]: As you say and according to definition:
`from which to accept requests`

Ethereumgo is the one who accept requests

File: docker-compose/mainnet/eth.yml (hunk 1 above)
Comment (phoenix-web3) [reply docker-compose/mainnet/eth.yml:5]: As you say and according to definition:
`from which to accept requests`

Ethereumgo is the one who accept requests

**Comment by phoenix-web3** [comment]
@swapper-pegasus Should we merge #23 and rebase before merging this PR?

**Comment by 0xp3gasus** [comment]
> @swapper-pegasus Should we merge #23 and rebase before merging this PR?

@swapper-phoenix Yes, we should merge #23 first, and then merge this pull request with version 1.1.1

**Review by 0xp3gasus — APPROVED** [review]

**Pull request merged**